internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -553,6 +636,103 @@ func (r *BMACReconciler) findBMH(ctx context.Context, agent *adiiov1alpha1.Agent
 	return nil, nil
 }
 
+func (r *BMACReconciler) ensureSpokeBMH(ctx context.Context, spokeClient client.Client, bmh *bmh_v1alpha1.BareMetalHost, machine *machinev1beta1.Machine) (*bmh_v1alpha1.BareMetalHost, error) {
+	bmhSpoke, mutateFn := r.newSpokeBMH(bmh, machine)
+	if result, err := controllerutil.CreateOrUpdate(ctx, spokeClient, bmhSpoke, mutateFn); err != nil {
+		r.Log.WithError(err).Errorf("failed to ensure spoke BareMetalHost %s/%s", bmhSpoke.Namespace, bmhSpoke.Name)
+		return nil, err
+	} else if result != controllerutil.OperationResultNone {
+		r.Log.Info("Spoke BareMetalHost created")
+	}
+	return bmhSpoke, nil
+}
+
+func (r *BMACReconciler) ensureSpokeMachine(ctx context.Context, spokeClient client.Client, bmh *bmh_v1alpha1.BareMetalHost, clusterDeployment *hivev1.ClusterDeployment) (*machinev1beta1.Machine, error) {
+	machineSpoke, mutateFn := r.newSpokeMachine(bmh, clusterDeployment)
+	if result, err := controllerutil.CreateOrUpdate(ctx, spokeClient, machineSpoke, mutateFn); err != nil {
+		r.Log.WithError(err).Errorf("failed to ensure spoke Machine %s/%s", machineSpoke.Namespace, machineSpoke.Name)
+		return nil, err
+	} else if result != controllerutil.OperationResultNone {
+		r.Log.Info("Spoke Machine created")
+	}
+	return machineSpoke, nil
+}
+
+func (r *BMACReconciler) ensureSpokeBMHSecret(ctx context.Context, spokeClient client.Client, bmh *bmh_v1alpha1.BareMetalHost) (*corev1.Secret, error) {
+	secretName := bmh.Spec.BMC.CredentialsName
+	secret := &corev1.Secret{}
+	err := r.Get(ctx, types.NamespacedName{Namespace: bmh.Namespace, Name: secretName}, secret)
```

Comment:
Now that this function is using `CreateOrUpdate` do you think it still needs to check for existing secrets? Perhaps it's an optimization we don't need? 

If the resource exists, then the `CreateOrUpdate` status should be `unchanged`

Reply:
It's checking for bmc secret on the hub cluster, and making a spoke copy of it in 	_newSpokeBMHSecret(secret, bmh)_
  and doing _CreateOrUpdate_ for the spoke bmc secret. In other words, it's needed so we can create the secret object on the spoke cluster.